src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,58 @@
+import classnames from 'classnames';
+import { noOp, px } from '../../utils';
+import CloseIcon from '../../assets/close.svg?react';
+import Button from '../Button/Button';
+import ReactModal from 'react-modal';
+
+export interface ModalProps extends ReactModal.Props {
+  /**
+   * Boolean to determine if the modal is open
+   */
+  isOpen: boolean;
+  /**
+   * Function to close the modal
+   */
+  onClose?: () => void;
+}
+
+/**
+ * ## Overview
+ *
+ * A component for displaying a modal.
+ *
+ */
+
+const Modal = ({ children, className, isOpen = false, onClose = noOp, ...props }: ModalProps) => {
+  if (!isOpen) {
+    return null;
+  }
+
+  const buttonClick = (e: React.MouseEvent) => {
+    e.stopPropagation();
```

Comment:
why do we need to stopPropagation here?  Parent components just might wanna listen